### DISCUSSION TITLE
How to resolve relative path after deno compile?

### DISCUSSION THREAD

**Discussion by cawa-93** [denoland/deno#17922] [Q&A] [answered]
I have script that accept filepath, and run several additional binaries.

```ts
// mainProgram.ts
const datafile = Deno.args[0];

const subProgramPath = new URL('subProgram.exe', import.meta.url)

Deno.run({
    cmd: [subProgramPath, datafile, '--some-args']
})
```

I want to compile this script into an executable `mainProgram.exe` file so that I can easily distribute it to my users. 

The `subProgram.exe` always lies next to the `mainProgram.ts`. In same dirrectory.

The problem is that after compilation, the `subProgramPath` has hardcoded value and equals to compilation-machine paths. And it doesn't work on any of my users' machines. 

How I should resolve path to `./subProgram.exe` from `mainProgram.exe` after compilation? How to resolve path in runtime relativetly to `mainProgram.exe` location?

I should note that my users can simply drag and drop a `datafile` on the `mainProgram.exe`. This way, the `mainProgram.exe` will run, but the cwd will correspond to the location of the `datafile`, not the `mainProgram.exe`

**Comment by zachauten** [comment, marked as answer]
`Deno.execPath()` return the path of the compiled script when you use `deno compile` (when using `deno run`  it returns the path of  the deno executable).

**Reply by KaKi87** [reply]
Hi,

Because of this diverse behavior, `Deno.execPath()` is only useful is prod, and the one that's useful in dev is `import.meta.dirname`, so how to conditionally use one or the other depending on whether the script is compiled ?

Thanks
